lib/ecto/adapters/mysql.ex

```diff
@@ -119,7 +145,7 @@ defmodule Ecto.Adapters.MySQL do
   def loaders(:float, type),          do: [&float_decode/1, type]
   def loaders(:binary_id, type),      do: [Ecto.UUID, type]
   def loaders({:embed, _} = type, _), do: [&json_decode/1, &Ecto.Adapters.SQL.load_embed(type, &1)]
-  def loaders(_, type),               do: [type]
+  def loaders(type_a, type_b),        do: additional_loaders(type_a, type_b)
```

Comment:
All the loaders are inlined for the reasons of efficiency. This code is called very often, even a function call is a big overhead here - we should avoid it, if we can.